pkg/controller/certificaterequests/sync_test.go

```diff
@@ -180,8 +180,172 @@ func TestSync(t *testing.T) {
+								LastTransitionTime: &nowMetaTime,
+							}),
+							gen.SetCertificateRequestStatusCondition(cmapi.CertificateRequestCondition{
+								Type:               cmapi.CertificateRequestConditionReady,
+								Status:             cmmeta.ConditionFalse,
+								Reason:             "RequestDenied",
+								Message:            "The CertificateRequest was denied by an approval controller",
+								LastTransitionTime: &nowMetaTime,
+							}),
+							gen.SetCertificateRequestFailureTime(nowMetaTime),
+						),
+					)),
+				},
+			},
+		},
+		"should update RequestDenied if certificate request is denied and also has Failed condition": {
+			certificateRequest: gen.CertificateRequestFrom(baseCRNotApproved,
+				gen.SetCertificateRequestStatusCondition(cmapi.CertificateRequestCondition{
+					Type:               cmapi.CertificateRequestConditionDenied,
+					Status:             cmmeta.ConditionTrue,
+					Reason:             "Foo",
+					Message:            "Certificate request has been denied by cert-manager.io",
+					LastTransitionTime: &nowMetaTime,
+				}),
+				gen.SetCertificateRequestStatusCondition(cmapi.CertificateRequestCondition{
+					Type:               cmapi.CertificateRequestConditionReady,
+					Status:             cmmeta.ConditionFalse,
+					Reason:             "Failed",
+					Message:            "Request has failed",
+					LastTransitionTime: &nowMetaTime,
+				}),
+				gen.SetCertificateRequestFailureTime(nowMetaTime),
+			),
+			builder: &testpkg.Builder{
+				CertManagerObjects: []runtime.Object{baseIssuer, baseCR},
+				ExpectedEvents: []string{
+					"Warning RequestDenied The CertificateRequest was denied by an approval controller",
+				},
+				ExpectedActions: []testpkg.Action{
+					testpkg.NewAction(coretesting.NewUpdateSubresourceAction(
+						cmapi.SchemeGroupVersion.WithResource("certificaterequests"),
+						"status",
+						gen.DefaultTestNamespace,
+						gen.CertificateRequestFrom(baseCRNotApproved,
+							gen.SetCertificateRequestStatusCondition(cmapi.CertificateRequestCondition{
+								Type:               cmapi.CertificateRequestConditionDenied,
+								Status:             cmmeta.ConditionTrue,
+								Reason:             "Foo",
+								Message:            "Certificate request has been denied by cert-manager.io",
+								LastTransitionTime: &nowMetaTime,
+							}),
+							gen.SetCertificateRequestStatusCondition(cmapi.CertificateRequestCondition{
+								Type:               cmapi.CertificateRequestConditionReady,
+								Status:             cmmeta.ConditionFalse,
+								Reason:             "RequestDenied",
+								Message:            "The CertificateRequest was denied by an approval controller",
+								LastTransitionTime: &nowMetaTime,
+							}),
+							gen.SetCertificateRequestFailureTime(nowMetaTime),
+						),
+					)),
+				},
+			},
+		},
+		"should do nothing if request has both Denied and Ready with reason RequestDenied conditions": {
```

Comment:
this text also seems slightly confusing - maybe "should do nothing if request has a Denied condition and already has a False Ready condition"